packages/kbn-i18n/src/core.js

```
@@ -0,0 +1,164 @@
+
+const showError = error => {
+  if (process.env.NODE_ENV !== 'production') {
+    console.error(error);
+  }
+};
+
+const getMessageFormat = memoizeIntlConstructor(IntlMessageFormat);
+
+export default class I18n {
+  constructor(messages = {}, locale = messages.locale || EN_LOCALE) {
+    this.currentLocale = normalizeLocale(locale);
+    this.messages = { [this.currentLocale]: messages };
+    this.defaultLocale = EN_LOCALE;
+    this.formats = {};
+
+    if (messages.localeData) {
+      addLocaleData(messages.localeData);
+    }
+  }
+
+  addMessages(messages = {}, locale = messages.locale || EN_LOCALE) {
+    const normalizedLocale = normalizeLocale(locale);
+
+    this.messages[normalizedLocale] = {
+      ...this.messages[normalizedLocale],
+      ...messages,
+    };
+
+    if (messages.localeData) {
+      addLocaleData(messages.localeData);
+    }
+  }
+
+  getMessages() {
+    return this.messages[this.currentLocale];
+  }
+
+  setLocale(locale) {
+    this.currentLocale = normalizeLocale(locale);
+  }
+
+  getLocale() {
+    return this.currentLocale;
+  }
+
+  setDefaultLocale(locale) {
+    this.defaultLocale = normalizeLocale(locale);
+  }
+
+  getDefaultLocale() {
+    return this.defaultLocale;
+  }
+
+  defineFormats(formats) {
+    this.formats = formats;
+  }
+
+  getFormats() {
+    return this.formats;
+  }
+
+  translate(id, { values = {}, defaultMessage = '' } = {}) {
+    if (!id) {
+      showError('[I18n] An `id` must be provided to translate a message.');
+    }
+
+    const message = getMessageById(this.getMessages(), id);
+
+    if (!hasValues(values) && process.env.NODE_ENV === 'production') {
+      return message || defaultMessage || id;
+    }
+
+    let formattedMessage;
+
+    if (message) {
+      try {
+        const msg = getMessageFormat(
+          message,
+          this.getLocale(),
+          this.getFormats()
+        );
+
+        formattedMessage = msg.format(values);
+      } catch (e) {
+        showError(
```

__question:__ I'm wondering if we shouldn't swallow error when we had message for the specified locale, but failed to format it. Looks like it's a way too tolerant to developer/translator mistakes. What do you think?